internal/lsp/examples/index.json

```diff
@@ -0,0 +1,17 @@
+{
```

Comment:
This file exists in JSON so that it can be periodically updated based on the sitemap. I have code to generate this JSON which will come later.

```
curl -L https://docs.styra.com/sitemap.xml | xq . - > output.json

cat output.json | opa eval 'data.play.symbols' -d play.rego --format=raw --stdin-input | jq
```


play.rego

```rego
package play

import rego.v1

symbols := {"keywords": _keywords, "builtins": _builtins}

_keywords[name] := path if {
	some p in _pages
	p[0] == "keywords"

	name := p[1]

	path := concat("/", p)
}

_builtins[name] := path if {
	some p in _pages
	p[0] == "builtins"

	l := count(p)

	l == 2

	name := p[1]

	not count({p |
		some p in _pages
		p[0] == "builtins"
		p[1] == name
	}) > 1

	path := concat("/", p)
}

_builtins[name] := path if {
	some p in _pages
	p[0] == "builtins"

	l := count(p)

	l > 2

	name := concat(
		".",
		[
			replace(p[1], "_", "."),
			concat(".", array.slice(p, 2, l)),
		],
	)

	path := concat("/", p)
}

_prefix := "https://docs.styra.com/opa/rego-by-example/"

_pages contains page if {
	some i
	url := input.urlset.url[i].loc

	startswith(url, _prefix)

	page := split(trim_prefix(url, _prefix), "/")
}

```

Reply:
👍 There's some Rego in `build` already that's meant to be run periodically, but nothing to enforce that. Might be a good time to look into when you add this later.

Reply:
I was planning to add this automation in later, but shared the script here for reference when that time comes. since the content is quite static.